s2spy/utils.py

```diff
@@ -82,21 +84,66 @@ def check_input_frequency(calendar, data):
     """
     if isinstance(data, PandasData):
         data_freq = pd.infer_freq(data.index)
-        if data_freq is None: # Manually infer the frequency
+        if data_freq is None:  # Manually infer the frequency
             data_freq = np.min(data.index.values[1:] - data.index.values[:-1])
     else:
         data_freq = xr.infer_freq(data.time)
-        if data_freq is None: # Manually infer the frequency
+        if data_freq is None:  # Manually infer the frequency
             data_freq = (data.time.values[1:] - data.time.values[:-1]).min()
 
     if isinstance(data_freq, str):
         data_freq.replace("-", "")
-        if not re.match(r'\d+\D', data_freq):
-            data_freq = '1' + data_freq
+        if not re.match(r"\d+\D", data_freq):
+            data_freq = "1" + data_freq
 
     if pd.Timedelta(calendar.freq) < pd.Timedelta(data_freq):
         warnings.warn(
             """Target frequency is smaller than the original frequency.
             The resampled data will contain NaN values, as there is no data
             available within all intervals."""
         )
+
+
+def plot_interval(
+    anchor_date: pd.Timestamp,
+    interval: pd.Interval,
+    ax: plt.Axes,
+    color: str,
+    add_freq: bool = False,
+):
+    """Utility for the calendar visualization.
+
+    Plots a rectangle representing a single interval.
+
+    Args:
+        anchor_date: Pandas timestamp representing the anchor date.
+        interval: Interval that should be added to the plot.
+        ax: Axis to plot the interval in.
+        color: (Matplotlib compatible) color that the rectangle should have.
+        add_freq: Toggles if the frequency will be displayed on the rectangle
+    """
+    right = (anchor_date - interval.right).days
+    hwidth = (interval.right - interval.left).days
+
+    ax.add_patch(
+        Rectangle(
+            (right, anchor_date.year - 0.4),
+            hwidth,
+            0.8,
```

Comment:
If we provide user the freedom to choose how many anchor years to display, this height must be adaptive as well.

Reply:
This height has the unit of years. I.e. it scales with the y-axis limits. You can try it out when you only have, e.g. a single anchor year: it plots fine (using the full height of the plot).